tools/stats-definition-exporter/main_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package main
+
+import (
+	"bytes"
+	"log"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRegisterStats(t *testing.T) {
+	globalStats, dbStats, err := registerStats()
+
+	assert.NotNil(t, globalStats)
+	assert.NotNil(t, dbStats)
+	assert.NoError(t, err)
+}
+
+func TestFileOutput(t *testing.T) {
+	defer func() {
```

Comment:
I think the typical way to do testing on files is tmpfiles?

Reply:
Will review this more closely tomorrow